opcua/common/subscription.py

```diff
@@ -201,22 +201,54 @@ def _get_node(self, nodeid):
     def _get_filter_from_event_type(self, eventtype):
         eventtype = self._get_node(eventtype)
         evfilter = ua.EventFilter()
-        for property in get_event_properties_from_type_node(eventtype):
+        evfilter.SelectClauses = self._select_clauses_from_evtype(eventtype)
+        evfilter.WhereClause = self._where_clause_from_evtype(eventtype)
+        return evfilter
+
+    def _select_clauses_from_evtype(self, evtype):
+        clauses = []
+        for property in get_event_properties_from_type_node(evtype):
             op = ua.SimpleAttributeOperand()
-            op.TypeDefinitionId = eventtype.nodeid
+            op.TypeDefinitionId = evtype.nodeid
             op.AttributeId = ua.AttributeIds.Value
             op.BrowsePath = [property.get_browse_name()]
-            evfilter.SelectClauses.append(op)
-        return evfilter
-
-    def subscribe_events(self, sourcenode=ua.ObjectIds.Server, evtype=ua.ObjectIds.BaseEventType):
+            clauses.append(op)
+        return clauses
+
+    def _where_clause_from_evtype(self, evtype):
+        cf = ua.ContentFilter()
+        el = ua.ContentFilterElement()
+        # operands can be ElementOperand, LiteralOperand, AttributeOperand, SimpleAttribute
+        op = ua.SimpleAttributeOperand()
+        op.AttributeId = ua.AttributeIds.NodeId
+        el.FilterOperands.append(op)
+        for subtypeid in [st.nodeid for st in self._get_subtypes(evtype)]:
+            op = ua.LiteralOperand()
+            op.Value = subtypeid
+            el.FilterOperands.append(op)
+        el.FilterOperator = ua.FilterOperator.InList
+
+        cf.Elements.append(el)
+        return cf
+
+    def _get_subtypes(self, parent, nodes=None):
```

Comment:
Is getting subtypes needed anywhere else? If yes than you could add it into Node class, since it is on the end node operation